gap/base/projective.gi

```diff
@@ -16,50 +16,48 @@
 ##
 #############################################################################
 
+# A non-empty square matrix <m> over a field is a projective one iff it is a
+# non-zero scalar multiple of the identity matrix, i.e. iff it is a non-zero
+# scalar matrix.
 InstallGlobalFunction( IsOneProjective,
-  function(el)
-    local s, n, i, j, zero;
-    n := NrRows(el);
-    Assert(0, n = NrCols(el));
-    s := el[1,1];
-    if IsZero(s) then return false; fi;
-    zero := Zero(s);
-    for i in [1..n] do
-        if el[i,i] <> s then return false; fi;
-        for j in [1..n] do
-            if i <> j and el[i,j] <> zero then return false; fi;
-        od;
-    od;
-    return true;
+  function(m)
+    Assert(2, NrRows(m) = NrCols(m) and not IsEmpty(m));
+    return not IsZero(m[1,1]) and RECOG.IsScalarMat(m) <> false;
   end );
 
+# Two invertible matrices <a> and <b> over the same field define equal
+# projective elements, or are "equal modulo scalars", iff there exists a
+# (necessarily non-zero) scalar <s> in the field such that <s * a = b>.
 InstallGlobalFunction( IsEqualProjective,
-  function(a,b)
-    local p,s,i;
-    p := PositionNonZero(a[1]);
-    s := b[1,p];
-    if IsZero(s) then return false; fi;
-    s := s / a[1,p];
-    for i in [1..Length(a)] do
-        if s*a[i] <> b[i] then return false; fi;
-    od;
-    return true;
+  function(a, b)
+    local p, s;
+    Assert(2, NrRows(a) = NrRows(b) and not IsEmpty(a));
```

Comment:
`IsEmpty` is defined for collections, but MatrixObj are not generally collections (they don't support `Size` either, for example). Though I guess we might want to add something like `IsEmptyMatrix` or so to the MatrixObj API, with a default implementation that returns `NrCols(m)=0 or NrRows(m)=0`

Reply:
Ah right. I changed these sorts of checks to `NrRows(a) > 0`.